tedge/src/cli/connect/command.rs

```diff
@@ -176,7 +184,40 @@ where
 // the check can finish in the second try as there is no error response in the first try.
 
 #[tokio::main]
-async fn check_connection_c8y(port: u16) -> Result<(), ConnectError> {
+async fn check_connection_c8y(port: u16) -> Result<DeviceStatus, ConnectError> {
+    for i in 0..2 {
+        print!("Try {} / 2: Sending a message to Cumulocity. ", i + 1,);
+
+        let create_device_result = create_device(port);
+        match create_device_result.await {
+            Ok(DeviceStatus::New) => {
+                return Ok(DeviceStatus::New);
+            }
```

Comment:
Notice that this is actually never the case, since c8y send no response if the device is new.